### PR TITLE
Speed up file existence checks where feasible.

### PR DESCRIPTION
**Pull request by ADmad**
file_exists() is a lot slower than is_file() / is_dir().
Refs: https://bugs.php.net/bug.php?id=78285

`is_file()` doesn't check if the file is actually accessible but if there are file permission issues one if going to have problems one way or another anyway.

### CODE REVIEW

File: src/Core/Configure.php [hunk 1]
```diff
@@ -421,14 +421,10 @@ public static function version(): string
         }
 
         $path = dirname(dirname(__DIR__)) . DIRECTORY_SEPARATOR . 'config/config.php';
-        if (file_exists($path)) {
```
Comment (ADmad) [inline src/Core/Configure.php:424]: I don't see why this existence check was necessary. The `config/config.php` file is always going to exist in the cakephp package.

**Comment by othercorey** [comment]
Looks ok. The check in CacheEngine doesn't seem to need file_exists, but I'm not sure what that does does exactly.

File: src/Core/Configure.php (hunk 1 above)
Comment (dereuromark) [reply src/Core/Configure.php:424]: Wasnt that because of phinx or sth?

Comment (dereuromark) [reply src/Core/Configure.php:424]: Came from https://github.com/cakephp/cakephp/pull/13942

Comment (ADmad) [reply src/Core/Configure.php:424]: Hmm.. Right, it could be because of split packages.

Comment (ADmad) [reply src/Core/Configure.php:424]: I reinstated the check and just changed the function.

**Pull request merged**